apps/rush-lib/src/cli/actions/ListAction.ts

```diff
@@ -21,47 +29,73 @@ export class ListAction extends BaseRushAction {
     });
   }
 
+  public setJsonObject(name: string, version: string, path: string): IJsonEntity {
+    return {name: name, version: version, path: path};
+  }
+
   protected onDefineParameters(): void {
     this._version = this.defineFlagParameter({
       parameterLongName: '--version',
       parameterShortName: '-v',
-      description: 'If this flag is specified, the project version will be ' +
+      description:
+        'If this flag is specified, the project version will be ' +
         'displayed in a column along with the package name.'
     });
 
     this._path = this.defineFlagParameter({
       parameterLongName: '--path',
       parameterShortName: '-p',
-      description: 'If this flag is specified, the project path will be ' +
+      description:
+        'If this flag is specified, the project path will be ' +
         'displayed in a column along with the package name.'
     });
 
     this._fullPath = this.defineFlagParameter({
       parameterLongName: '--full-path',
       parameterShortName: '-f',
-      description: 'If this flag is specified, the project full path will ' +
+      description:
+        'If this flag is specified, the project full path will ' +
         'be displayed in a column along with the package name.'
     });
+
+    this._jsonFlag = this.defineFlagParameter({
+      parameterLongName: '--json',
+      description: 'If specified, output will be in JSON format.'
+    });
   }
 
   protected run(): Promise<void> {
     return Promise.resolve().then(() => {
       const allPackages: Map<string, RushConfigurationProject> = this.rushConfiguration.projectsByName;
       if (this._version.value || this._path.value || this._fullPath.value) {
         this._printListTable(allPackages);
+      } else if (this._jsonFlag.value) {
+        this._serializeToJson(allPackages);
       } else {
         this._printList(allPackages);
       }
     });
   }
 
+  private _serializeToJson(
+    allPackages: Map<string, RushConfigurationProject>
+  ): void {
+    let i: number = 0;
+    allPackages.forEach((_config: RushConfigurationProject, name: string) => {
+     this._objects[i++] = JSON.stringify(this.setJsonObject(name, _config.packageJson.version,  _config.projectFolder));
```

Comment:
This will end up serialized as an array of serialized strings. We want a full data structure in a single serialized block.

What you have looks like this:
```json
[ '{"name":"@microsoft/api-documenter","version":"7.7.2","path":"C:\\\\code\\\\rushstack\\\\apps\\\\api-documenter"}',
  '{"name":"@microsoft/api-extractor","version":"7.7.0","path":"C:\\\\code\\\\rushstack\\\\apps\\\\api-extractor"}',
  '{"name":"@microsoft/api-extractor-model","version":"7.7.0","path":"C:\\\\code\\\\rushstack\\\\apps\\\\api-extractor-model"}',
  ... ]
```

It should probably look like this:

```json
{
  "projects": [
    {
      "name": "@microsoft/api-documenter",
      "version": "7.7.2",
      "path": "apps/api-documenter",
      "fullPath": "C:\\code\\rushstack\\apps\\api-documenter"
    },
    {
      "name": "@microsoft/api-extractor",
      "version": "7.7.0",
      "path": "apps/api-extractor",
      "fullPath": "C:\\code\\rushstack\\apps\\api-extractor"
    },
    {
      "name": "@microsoft/api-extractor-model",
      "version": "7.7.0",
      "path": "apps/api-extractor-model",
      "fullPath": "C:\\code\\rushstack\\apps\\api-extractor-model"
    },
    ...
  ]
}
```